packages/provider/index.js

```diff
@@ -17,7 +17,9 @@ module.exports = {
   getProvider: function (options) {
     let provider;
     if (options.provider && typeof options.provider === "function") {
+      console.log("Unthunking provider:", ++count);
       provider = options.provider();
+      options.provider = provider;
```

Comment:
Cache the provider, so there's only ever one for the duration of the process.

Reply:
oh yeaaahhh

Reply:
side effect: the user _wants_ it to be called a bajillion times, and now it doesn't.

I think this was a bug in Truffle, but it's been around for so long users may be relying on it.

Reply:
Well, no user was relying on it because they didn't have access to the functionality because webpack/original-require didn't allow it :)